Add API and experimetal templates (#4258)
opensearch-project/documentation-website: API_STYLE_GUIDE.md

# API Style Guide

> This guide provides the basic structure for creating OpenSearch API documentation. It includes the most important elements that should appear in the API documentation and their descriptions.

Reviewer: ```suggestion
This guide provides the basic structure for creating OpenSearch API documentation. It includes the various elements that we feel are most important to creating complete and useful API documentation, as well as description and examples where appropriate.
```

> # API Style Guide

Reviewer 2: Vale catch.

```suggestion
# API style guide
```

Author: Yeah, in this case I think Vale is wrong :)

> This guide provides the basic structure for creating OpenSearch API documentation. It includes the most important elements that should appear in the documentation and helpful suggestions to help support them. 

Author: Reworded :)

Depending on the intended purpose of the API, *some sections will be required while others may not be applicable*.

Use the [API_TEMPLATE](templates/API_TEMPLATE.md) to create an API documentation page.

### A note on terminology ###

> Terminology for API parameters varies in the software industry, where two or even three names may be used to label the same type of parameter. For the sake of consistency, we use the following nomenclature for parameters in our API documentation:

Reviewer 3: ```suggestion
Terminology for API parameters varies in the software industry, where two or even three names may be used to label the same type of parameter. For consistency, we use the following nomenclature for parameters in our API documentation:
```

## Basic elements for documentation

> The following sections describe the basic API documentation structure. Each section is discussed under its respective heading. You can include only those elements appropriate to the API. 

Reviewer 3: ```suggestion
The following sections describe the basic API documentation structure. Each section is discussed under its respective heading. Include only those elements appropriate to the API. 
```

```json
GET _search/scroll/<scroll_id>
```

> In the preceding example, the resource is `scroll` and its path parameter is `<scroll_id>`.

Reviewer 2: Nit, should we move the description of the example above the example?

GET _cat/aliases?v
```

> In the preceding example, the endpoint is `aliases` and its query parameter is `v` (provides verbose output).

Reviewer 2: See comment on line 78.
